### PR TITLE
Add Teleop to the Robot

### PR DESCRIPTION
**Pull request by a2aaron**
Blocked by PR #12 (Add Gamepad to the Robot)

### CODE REVIEW

File: src/org/usfirst/frc/team1076/robot/Robot.java [hunk 1]
```diff
@@ -39,10 +47,11 @@
 	 */
 	@Override
 	public void robotInit() {
+	    Strongback.start();
 		oi = new OI();
-		chooser.addDefault("Default Auto", new ExampleCommand());
-		// chooser.addObject("My Auto", new MyAutoCommand());
-		SmartDashboard.putData("Auto mode", chooser);
+		gamepad = new Gamepad(0);
+        SmartDashboard.putNumber("Left Factor", 1);
```
Comment (saltyskittlez) [inline src/org/usfirst/frc/team1076/robot/Robot.java]: Use the Smarter Dashboard; put default.

File: src/org/usfirst/frc/team1076/robot/commands/TeleopCommand.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	IGamepad gamepad;
+	
+    public TeleopCommand(IGamepad gamepad, Drivetrain leftRight ) {
+         super(leftRight); //Require the motors
+         this.gamepad = gamepad;
+         this.leftRight = leftRight;
+    }
+
+    // Called repeatedly when this Command is scheduled to run
+    public boolean execute() {
```
Comment (saltyskittlez) [inline src/org/usfirst/frc/team1076/robot/commands/TeleopCommand.java:26]: This method should be marked `@Override`

File: src/org/usfirst/frc/team1076/robot/commands/TeleopCommand.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    	// is 0.5, then we'll get 2.0 and divide by 2.0.
+    	final double norm = selectMaxAbs(new double[] {1/maxSpeed, left, right});
+    	
+    	leftRight.setLeftSpeed(left / norm);
+    	leftRight.setRightSpeed(right / norm);
+    	
+    	return isFinished();
+    }
+    
+    private double selectMaxAbs(double[] items) {
```
Comment (saltyskittlez) [inline src/org/usfirst/frc/team1076/robot/commands/TeleopCommand.java]: Use the `double...` type.

File: src/org/usfirst/frc/team1076/robot/subsystems/Drivetrain.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	
+	/**
+	 * Stops left and right motors.
+	 */
+	public void stop() {
+		setSpeed(0);
+	}
+
+    public void initDefaultCommand() {
```
Comment (saltyskittlez) [inline src/org/usfirst/frc/team1076/robot/subsystems/Drivetrain.java]: Do we need this code anymore?

File: testsrc/org/usfirst/frc/team1076/test/TestTeleopCommand.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+				1.0, left.getSpeed(), EPSILON);
+		assertEquals("Right motor should be 1.0 when going forward",
+				1.0, right.getSpeed(), EPSILON);
+	}
+	
+	@Test
+	public void testRotate() {
+		gamepad.lx = 1;
+		teleop.execute();
+		assertEquals("Left motor should be negative when turning left",
```
Comment (saltyskittlez) [inline testsrc/org/usfirst/frc/team1076/test/TestTeleopCommand.java]: Which direction is the motor moving in?

**Review by saltyskittlez — CHANGES_REQUESTED** [review]
`Put default` is a critical change.

File: src/org/usfirst/frc/team1076/robot/Robot.java (hunk 1 above)
Comment (a2aaron) [reply src/org/usfirst/frc/team1076/robot/Robot.java]: This can be done when we merge the Smarter Dashboard PR

**Review by saltyskittlez — APPROVED** [review]
👍

**Pull request merged**